### PR TITLE
Add more classifiers

### PR DESCRIPTION
**Pull request by useanalias**
Hello again!

I've recently implemented the following classifiers into hyperopt-sklearn:

- AdaBoost
- Single decision tree
- Gaussian Naive Bayes
- Gradient Boosting
- Linear Discriminant Analysis
- Quadratic Discriminant Analysis
- Passive Aggressive Online Learning

I hope this helps! Let me know if I should change anything.

Many thanks,
Chandler

### CODE REVIEW

**Comment by bjkomer** [comment]
Looks great! Thanks again for all of these improvements. Did some quick tests and everything seems to run fine except for one minor issue. 

For LDA, the 'shrinkage' parameter is not supported with the default solver of 'svd' (found out about this from [this](https://github.com/scikit-learn/scikit-learn/issues/5354) issue). We'll have to make sure that particular parameter combination is never chosen. The way I've done this in the past is to make a combined choice parameter that contains all possible valid sets of the parameters. It's not very pretty, but it seems to work. I'm open to suggestions of alternative ways of handling these cases. An example of this is `liblinear_svc` around line [376](https://github.com/hyperopt/hyperopt-sklearn/blob/master/hpsklearn/components.py#L376) in `components.py`

**Comment by useanalias** [comment]
Thank you! I'll mirror what you did and get a fix pushed today.

**Comment by useanalias** [comment]
I believe that fixed the bug - thanks for letting me know!

**Comment by bjkomer** [comment]
Looks good! Only thing I'd change is to remove the `_dual` from the name of the combined parameter. The reason it was there in the `liblinear_svc` case was because there were three parameters, the third of which was called `dual`.

**Comment by useanalias** [comment]
Whoops! Thanks for letting me know - that should do it. :smile:
